Security/Firewall/OAuth2Listener.php

```diff
@@ -68,7 +68,7 @@ public function __invoke(RequestEvent $event)
             /** @var OAuth2Token $authenticatedToken */
             $authenticatedToken = $this->authenticationManager->authenticate($this->oauth2TokenFactory->createOAuth2Token($request, null, $this->providerKey));
         } catch (AuthenticationException $e) {
-            throw Oauth2AuthenticationFailedException::create($e->getMessage());
+            throw new Oauth2AuthenticationFailedException($e->getMessage(), 401, $e);
```

Comment:
@Orkin please reuse the current create method. add a `?Throwable $previous = null` parameter to it.

Reply:
@X-Coder264 Changing the signature of the `change` method would be a breaking change.

Reply:
Strictly technically speaking yes it is, but the chance of somebody overriding it is very slim IMHO. Ok, I'll merge this as is then.

btw, it's `create`, not `change` :smile: